### DISCUSSION TITLE
Afore HNS8000TL integration

### DISCUSSION THREAD

**Discussion by BigBoy1760** [davidrapan/ha-solarman#257] [Q&A]
Hi All,

First, I want to say a huge thanks for all the effort and everything you are doing to maintain such a database of different inverters. I have been trying to integrate my Afore HNS8000TL inverter with my Home Assistant (HA) since I got it.

I believe my logger is a Solarman 350xx (Module Version No: LSW5BLE_17_1E03_1.20R). When I add the integration to HA, it is automatically picked up by the Autodiscover feature, which means there is some sort of communication. However, the integration could not initialize correctly. From the logs, I can see the following error: "V5FrameError: V5 frame does not contain a valid Modbus RTU frame".

I have used "afore_2mppt.yaml" as the config file. My question is: Is the problem due to my 350xx device? As I have read that this might be the case, is there something I can run from my side to troubleshoot the issue further?

[solarman_logs.txt](https://github.com/user-attachments/files/17688513/solarman_logs.txt)

Thanks in advance.

Regards,
Bobby

**Comment by davidrapan** [comment]
Hi @BigBoy1760, did you also tried the other [afore_BNTxxxKTL-2mppt.yaml](https://github.com/davidrapan/ha-solarman/blob/main/custom_components/solarman/inverter_definitions/afore_BNTxxxKTL-2mppt.yaml) profile?

The error you see simply means that you are asking the inverter for values of incorrect registers. So as you correctly assumed integration does communicate with your inverter as it should and it's only matter of profile with correct registers.

You could ask your manufacturer for the modbus documentation for your device. (Maybe they will be willing to)

But without the doc the only option is to try and guess which is really tedious...

**Comment by murik85** [comment]
HNS seems to have a new and an old version. If new version try [afore_hybrid.yaml](https://github.com/davidrapan/ha-solarman/blob/main/custom_components/solarman/inverter_definitions/afore_hybrid.yaml).

**Reply by davidrapan** [reply]
[afore_2mppt.yaml](https://github.com/davidrapan/ha-solarman/blob/main/custom_components/solarman/inverter_definitions/afore_2mppt.yaml) is basically [afore_hybrid.yaml](https://github.com/davidrapan/ha-solarman/blob/main/custom_components/solarman/inverter_definitions/afore_hybrid.yaml) but w/o Battery sensors and such.

**Reply by murik85** [reply]
Then it should work.

**Reply by BigBoy1760** [reply]
Actually, I have tried with each and every profile prior to starting the discussion. I guess I will try finding the correct documentation.

Regarding the guessing part, maybe I can alter your code by adding some sort of loop in order to find the correct value.

Thanks for you answers.

**Reply by murik85** [reply]
Is the logger native or purchased separately?

**Reply by BigBoy1760** [reply]
native, it came with the inverter, in its html dashboard I can see data. 

![image](https://github.com/user-attachments/assets/c68977c7-eb5e-4006-8e38-b9045ce1f625)

**Reply by murik85** [reply]
Write to info@aforenergy.com and ask them to send you the documentation of the protocol.

**Reply by BigBoy1760** [reply]
Hi All,

Let's make a little bump on the topic. I have received the documentation from Afore and tried to make something on my own, which ended as a lot of wasted time (am not a coder). Currently the only two things that I have achieved is to connect with the inverter via the integration and not be able to read any sensor:

<img width="670" alt="{959BFA87-7B23-4D25-B5B2-8E00A897E6EE}" src="https://github.com/user-attachments/assets/c8337bcb-627a-4bd2-be92-76f7f63cfc74" />

Secondly connecting to the inverter directly using python, where I have received some data, but all seems to be gibberish....

I will upload a file with the errors form the HA integration (IP is masked...), no matter what I have tried I didn't have the time and knowledge to resolve them. 

For the YAML file I have used the chatbot to help me generate it, but didn't help as well. I have used different register values without any luck.

The other uploaded files will be the mentioned YAML and datasheet if they are helpful to anyone...

[HA_error_Logs.txt](https://github.com/user-attachments/files/19160752/HA_error_Logs.txt)

[Communication Protocol of S6 Single Phase Inverter v3.6_build00_211104.docx](https://github.com/user-attachments/files/19160692/Communication.Protocol.of.S6.Single.Phase.Inverter.v3.6_build00_211104.docx)

[afore_HNSxxxTL-2mppt.yaml.txt](https://github.com/user-attachments/files/19160755/afore_HNSxxxTL-2mppt.yaml.txt)

Regards,
Blagoy

**Reply by lucky5822** [reply]
@BigBoy1760 Any luck to connect with the inverter? I have the same model but I'm unable to read any data. I'm waiting from afore the modbus registers.

**Reply by BigBoy1760** [reply]
Unfortunately, no. It's above my knowledge. I have tried qModMaster and I can connect to the inverter, but could not get the slaveID and any date from it:

![image](https://github.com/user-attachments/assets/8641a7ce-eff9-43f7-a1cd-de07a5ad6213)

@lucky5822 does your WIFI dongle start with 350xxxx

Regards,
Blagoy

**Reply by lucky5822** [reply]
> Unfortunately, no. It's above my knowledge. I have tried qModMaster and I can connect to the inverter, but could not get the slaveID and any date from it:
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/8641a7ce-eff9-43f7-a1cd-de07a5ad6213)
> 
> 
> 
> @lucky5822 does your WIFI dongle start with 350xxxx
> 
> 
> 
> Regards,
> 
> Blagoy 

Same here, tried different software fo modbus polling but no answer from inverter.
My dongle starts with 352xxxxxxx
